cluster-autoscaler/processors/status/scale_down_status_processor.go

```diff
@@ -52,6 +52,23 @@ const (
 	ScaleDownNodeDeleted
 	// ScaleDownNodeDeleteStarted - a node deletion process was started.
 	ScaleDownNodeDeleteStarted
+	// ScaleDownNotTried - the scale down wasn't even attempted, e.g. an autoscaling iteration was skipped, or
+	// an error occurred before the scale up logic.
+	ScaleDownNotTried
+	// ScaleDownBlockedOnScaleUp - the scale down wasn't even attempted, because a scale-up was successful.
+	ScaleDownBlockedOnScaleUp
+	// ScaleDownDisabled - the scale down wasn't attempted, because it was disabled.
+	ScaleDownDisabled
+	// ScaleDownDeleteInProgress - the scale down wasn't attempted, because a node delete was still in progress.
+	ScaleDownDeleteInProgress
+	// ScaleDownScaleUpTooRecent - the scale down wasn't attempted, because a scale-up happened too recently.
+	ScaleDownScaleUpTooRecent
+	// ScaleDownFailureTooRecent - the scale down wasn't attempted, because a scale-down failure happened too recently.
+	ScaleDownFailureTooRecent
+	// ScaleDownNodeDeleteTooRecent - the scale down wasn't attempted, because a scale-down node delete happened too recently.
+	ScaleDownNodeDeleteTooRecent
+	// ScaleDownForbidden - the scale down wasn't attempted, because it was forbidden by previous logic.
+	ScaleDownForbidden
```

Comment:
This is leaking a huge part of internal implementation details to the API. Can we replace it with something more generic?

Reply:
Done